### PR TITLE
[9.x] Use match expression where possible

### PR DESCRIPTION
**Pull request by oliverquynh**
Laravel 9 will require PHP 8.0 as the minimum version, so we are free to use match expression. In this PR, I try to convert a few switch statements into match expressions that makes code easier to read.

Please review the code changes and see how nice the match expression is.

### CODE REVIEW

File: src/Illuminate/Auth/CreatesUserProviders.php [hunk 1]
```diff
@@ -33,16 +33,13 @@ public function createUserProvider($provider = null)
...
-                return $this->createEloquentProvider($config);
-            default:
-                throw new InvalidArgumentException(
-                    "Authentication user provider [{$driver}] is not defined."
-                );
-        }
+        return match ($driver) {
+            'database' => $this->createDatabaseProvider($config),
+            'eloquent' => $this->createEloquentProvider($config),
+            default => (throw new InvalidArgumentException(
```
Comment (taylorotwell) [inline src/Illuminate/Auth/CreatesUserProviders.php:39]: Why is the throw wrapped in parentheses?

Comment (seriquynh) [reply src/Illuminate/Auth/CreatesUserProviders.php]: There are not necessary. I removed them.

**Comment by driesvints** [comment]
@seriquynh this seem to have broken the default `null` return in a couple of places: https://github.com/laravel/framework/pull/40880

**Comment by seriquynh** [comment]
@driesvints I'm sorry for missing your mention because I haven't check github for a while. I see a fixed PR has been merged. Thanks for you guys!

**Pull request merged**